src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -2463,6 +2477,12 @@ private static unsafe void FreeArray(byte** arr, int length)
             int redirectStdin, int redirectStdout, int redirectStderr, int creationFlags,
             out int lpChildPid, out int stdinFd, out int stdoutFd, out int stderrFd);
 
+        [DllImport("libpsl-native", SetLastError = true)]
+        internal static extern bool KillProcess(int pid);
+
+        [DllImport("libpsl-native", SetLastError = true)]
+        internal static extern int WaitPid(int pid, bool nohang);
```

Comment:
Do we really need this in libpsl? Why not call from libc directly? Search by `DllImport("libc"` for examples.

Also we should add `[return: MarshalAs(UnmanagedType.Bool)]`.

Reply:
Signals and other constants don't have a defined, standardized value, so I prefer to wrap it in libpsl and use the header constants there rather than coding it into S.M.A.

> Also we should add `[return: MarshalAs(UnmanagedType.Bool)]`.

Should I do the same for WaitPid as well?

Reply:
I don't see we use these constants - I see only 2 parameters and return value.

Reply:
These are referenced in libpsl wrappers: PowerShell/PowerShell-Native#63

If you called directly into libc from S.M.A then you'd need to call `kill(pid, SIGKILL)` and `waitpid(pid, NULL, WNOHANG)`.

Reply:
Aren't both POSIX?

Reply:
POSIX doesn't define the specific values of either constants (though SIGKILL should equal 9 most of the time).

Reply:
Do they really different on supported platforms?

Reply:
I haven't looked into all platforms but following POSIX should be more portable than hardcoding the constants in S.M.A. While it might work with SIGKILL I doubt it'll be very reliable to hardcode WNOHANG. In addition, if we directly imported kill()/waitpid() in S.M.A it would then be possible to use them directly with nonportable constants, so I prefer to restrict the call interface in this fashion.